Extensions/Leaderboards/leaderboardstools.ts

```diff
@@ -45,74 +116,196 @@ namespace gdjs {
 
-        if (
-          (_lastPlayerName === playerName && _lastScore === score) ||
-          (!!_scoreLastSentAt && Date.now() - _scoreLastSentAt < 500)
-        ) {
-          errorVar.setString('Wait before sending a new score.');
-        } else {
-          const baseUrl = 'https://api.gdevelop-app.com/play';
-          const game = runtimeScene.getGame();
-          fetch(
-            `${baseUrl}/game/${gdjs.projectData.properties.projectUuid}/leaderboard/${leaderboardId}/entry`,
-            {
-              body: JSON.stringify({
-                playerName: formatPlayerName(playerName),
-                score: score,
-                sessionId: game.getSessionId(),
-                clientPlayerId: game.getPlayerId(),
-                location:
-                  typeof window !== 'undefined' && (window as any).location
-                    ? (window as any).location.href
-                    : '',
-              }),
-              method: 'POST',
-              headers: { 'Content-Type': 'application/json' },
+        const baseUrl = 'https://api.gdevelop-app.com/play';
+        const game = runtimeScene.getGame();
+        const formattedPlayerName = formatPlayerName(playerName);
+        const sessionId = game.getSessionId();
+        const playerId = game.getPlayerId();
+        const location =
+          typeof window !== 'undefined' && (window as any).location
+            ? (window as any).location.href
+            : '';
+        fetch(
+          `${baseUrl}/game/${gdjs.projectData.properties.projectUuid}/leaderboard/${leaderboardId}/entry`,
+          {
+            body: JSON.stringify({
+              playerName: formattedPlayerName,
+              score: score,
+              sessionId: sessionId,
+              clientPlayerId: playerId,
+              location: location,
+            }),
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+          }
+        ).then(
+          (response) => {
+            if (!response.ok) {
+              const errorCode = response.status.toString();
+              logger.error(
+                'Server responded with an error:',
+                errorCode,
+                response.statusText
+              );
+              scoreSavingData.setError(errorCode);
+              errorVar.setString(errorCode);
+              return;
             }
-          )
-            .then((response) => {
-              _scoreLastSentAt = Date.now();
-              if (!response.ok) {
-                errorVar.setString(response.status.toString());
-                _lastErrorCode = response.status;
-                return response.statusText;
-              } else {
-                _lastScore = score;
-                _lastPlayerName = playerName;
-                _lastErrorCode = response.status;
-                return response.text();
+
+            scoreSavingData.closeSaving();
+
+            response.text().then(
```

Comment:
Nit: 
```suggestion
            return response.text().then(
```

So that someone who would in the future await/`then` on the initial promise is not surprised about this not being finished before their new code is called.